homeassistant/components/ecoforest/switch.py

```diff
@@ -0,0 +1,75 @@
+"""Switch platform for Ecoforest."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pyecoforest.models.device import Device
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import EcoforestCoordinator
+from .entity import EcoforestEntity
+
+
+@dataclass
+class EcoforestRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[Device], bool]
+
+
+@dataclass
+class EcoforestSwitchEntityDescription(
+    SwitchEntityDescription, EcoforestRequiredKeysMixin
+):
+    """Describes an Ecoforest switch entity."""
+
+
+SWITCH_TYPES: tuple[EcoforestSwitchEntityDescription, ...] = (
+    EcoforestSwitchEntityDescription(
+        key="status", translation_key="status", value_fn=lambda data: data.on
```

Comment:
```suggestion
        key="status", name=None, value_fn=lambda data: data.on
```

Reply:
Can you clarify this change so I can understand the reasoning

Reply:
The reason for this btw, when you use `has_entity_name = True` (which you do) and you put name to None, you use the device name as entity name (Which is good since you are turning the fireplace on and off). So there is no real need for an extra translation key here

Reply:
> Which is good since you are turning the fireplace on and off
This part was intentional. 😄